crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -56,7 +58,26 @@ impl<'db> UnionBuilder<'db> {
         self
     }
 
-    pub(crate) fn build(self) -> Type<'db> {
+    /// Performs the following normalizations:
+    ///     - Replaces `Literal[True,False]` with `bool`.
+    ///     - TODO For enums `E` with members `X1`,...,`Xn`, replaces
+    ///     `Literal[E.X1,...,E.Xn]` with `E`.
+    fn simplify(&mut self) {
+        if self
+            .elements
+            .is_superset(&[Type::BooleanLiteral(true), Type::BooleanLiteral(false)].into())
+        {
+            let bool_ty = builtins_symbol_ty_by_name(self.db, "bool");
+            if !bool_ty.is_unbound() {
+                self.elements.remove(&Type::BooleanLiteral(true));
+                self.elements.remove(&Type::BooleanLiteral(false));
+                self.elements.insert(bool_ty);
+            }
```

Comment:
I would just eliminate this check; if someone uses a typeshed without a `bool` type, then `Unknown` is the right type here. (I think there's a separate TODO here, which is that in most cases where we look up a type from builtins, we should be replacing `Unbound` with `Unknown` -- but this code here shouldn't be worrying about that detail.)
```suggestion
            self.elements.remove(&Type::BooleanLiteral(true));
            self.elements.remove(&Type::BooleanLiteral(false));
            self.elements.insert(bool_ty);
```